### DISCUSSION TITLE
Autocomplete branches

### DISCUSSION THREAD

**Discussion by Enigama** [pwntester/octo.nvim#502] [Q&A]
When we create a PR is it possible to have autcomplete in input like we have it for reviewers?
I also would like to know is this possible? Technically I found [REST api](https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#list-branches) for it. Would be cool to do it my own but I'm not good at Lua at all.

![image](https://github.com/pwntester/octo.nvim/assets/28072229/8df465bd-9918-4c9e-88ab-b932d33c1542)

**Comment by Enigama** [comment]
@pwntester

**Comment by williambdean** [comment]
Hi @Enigama 

Can you give context to where you this would be?

**Comment by williambdean** [comment]
Would @MuntasirSZN know how to do this?

**Reply by MuntasirSZN** [reply]
And yes, its possible.

**Reply by williambdean** [reply]
Is it a change from our end? Would this require many changes?

**Reply by MuntasirSZN** [reply]
Yep, using a picker.

**Reply by MuntasirSZN** [reply]
Also not that many changes needed, we just need write some lua for the pickers.

**Reply by MuntasirSZN** [reply]
A query like this:

```sh
gh api \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/pwntester/octo.nvim/branches
```

Does it for branches.

**Reply by williambdean** [reply]
Thats fine. Where does that get triggered?

**Reply by MuntasirSZN** [reply]
Head branch.

**Reply by MuntasirSZN** [reply]
For base branches, we need to replace owner and repo with the forked owner and repo.

**Reply by MuntasirSZN** [reply]
Also, if you think writing for each picker is a pain, use vim.ui.select (i forgot to pr) in place of the input.

**Reply by Enigama** [reply]
Is there any updates? Or I can try to implement it

**Reply by williambdean** [reply]
There is some logic here: 

https://github.com/pwntester/octo.nvim/blob/0e3f9aec9089b4eebd0e4466af85bb135d423436/lua/octo/completion/search.lua?plain=1#L345-L346

However, that is used for searching of issues and PRs, not while making of the PRs. So any efforts for creation of PRs would be separate from that and apprecitated to understand better!
